core/parachain/validator/impl/parachain_processor.cpp

```diff
@@ -1051,10 +1107,17 @@ namespace kagome::parachain {
     handleNotify(peer_id, relay_parent);
   }
 
+  bool ParachainProcessorImpl::isValidatingNode() const {
+    BOOST_ASSERT(this_context_->get_executor().running_in_this_thread());
+    return (app_config_.roles().flags.authority == 1);
```

Comment:
Another condition is at least having keys, and better to ensure being of actual validator for current epoch.

Reply:
Keys are checked by [`ValidatorSignerFactory::at`](https://github.com/soramitsu/kagome/blob/a6cff17fec2ccbe9dd153a7120cf67ee82d306d5/core/parachain/validator/signer.hpp#L103-L105)